go/v1beta1/storage/elasticsearch_test.go

```diff
@@ -1608,8 +1807,34 @@ var _ = Describe("elasticsearch storage", func() {
 			Expect((*searchBody.Query.Term)["name"]).To(Equal(expectedNoteName))
 		})
 
-		It("should immediately refresh the index", func() {
-			Expect(transport.receivedHttpRequests[0].URL.Query().Get("refresh")).To(Equal("true"))
+		When(fmt.Sprintf("refresh configuration is %s", config.RefreshTrue), func() {
+			BeforeEach(func() {
+				esConfig.Refresh = config.RefreshTrue
+			})
+
+			It("should immediately refresh the index", func() {
+				Expect(transport.receivedHttpRequests[0].URL.Query().Get("refresh")).To(Equal("true"))
+			})
+		})
+
+		When(fmt.Sprintf("refresh configuration is %s", config.RefreshWaitFor), func() {
+			BeforeEach(func() {
+				esConfig.Refresh = config.RefreshWaitFor
+			})
+
+			It("should immediately refresh the index", func() {
+				Expect(transport.receivedHttpRequests[0].URL.Query().Get("refresh")).To(Equal("true"))
+			})
+		})
+
+		When(fmt.Sprintf("refresh configuration is %s", config.RefreshFalse), func() {
+			BeforeEach(func() {
+				esConfig.Refresh = config.RefreshFalse
+			})
+
+			It("should not wait or force refresh of index", func() {
+				Expect(transport.receivedHttpRequests[0].URL.Query().Get("refresh")).To(Equal("false"))
```

Comment:
thoughts on using the `RefreshOption` type here for the assertion so we aren't using magic strings?

Reply:
I wasn't sure either way. It would cut down on copy/paste magic strings, but then it is tying implementation to those consts, and that feels like copying production code into tests. It also just happens that I made those be the exact values to pass to refresh parameter, but I could have as well made it an `int` type and use `iota`. I ended up deciding to test directly against end state value.